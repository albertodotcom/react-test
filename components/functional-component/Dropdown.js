import React from 'react';
import PropTypes from 'prop-types';

const Dropdown = props => (
  <select
    className="t-select"
    value={props.selected}
    onChange={e => props.onChange(e.target.value)}
  >
    { props.options.map(o =>
      <option className="t-option" key={o.id} value={o.id}>{o.text}</option>)}
  </select>
);

Dropdown.propTypes = {
  options: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired,
  })).isRequired,
  selected: PropTypes.string,
  onChange: PropTypes.func.isRequired,
};

Dropdown.defaultProps = {
  selected: '',
};

export default Dropdown;
