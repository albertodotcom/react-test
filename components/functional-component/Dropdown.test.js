/* eslint-env jest */
import React from 'react';
import { shallow } from 'enzyme';

import Dropdown from './Dropdown';

describe('Dropdown', () => {
  const defaultProps = {
    options: [
      { id: '8', text: 'Conte' },
      { id: '10', text: 'Del Piero' },
      { id: '1', text: 'Buffon' },
    ],
    selected: '10',
    onChange: () => { },
  };

  describe('element testing', () => {
    it('renders a list of options', () => {
      const options = [
        { id: '10', text: 'Del Piero' },
        { id: '21', text: 'Dybala' },
      ];

      const component = shallow(
        <Dropdown
          {...defaultProps}
          options={options}
        />);

      const optionTags = component.find('.t-option');
      expect(optionTags.length).toBe(2);
      expect(optionTags.at(0).text()).toEqual(options[0].text);
      expect(optionTags.at(1).text()).toEqual(options[1].text);
    });

    it('displays the text for the selected value', () => {
      const component = shallow(
        <Dropdown
          {...defaultProps}
          selected="1"
        />);

      expect(component.find('.t-select').prop('value')).toEqual('1');
    });
  });

  describe('snapshot testing', () => {
    it('renders a list of options', () => {
      const options = [
        { id: '10', text: 'Del Piero' },
        { id: '21', text: 'Dybala' },
      ];

      const wrapper = shallow(
        <Dropdown
          {...defaultProps}
          options={options}
        />);

      expect(wrapper).toMatchSnapshot();
    });
  });

  it('calls onChange with the new value', () => {
    const onChange = jest.fn();

    const component = shallow(
      <Dropdown
        {...defaultProps}
        onChange={onChange}
      />);
    component.find('.t-select').simulate('change', { target: { value: '123' } });

    expect(onChange.mock.calls.length).toBe(1);
    expect(onChange).toBeCalledWith('123');
  });
});
