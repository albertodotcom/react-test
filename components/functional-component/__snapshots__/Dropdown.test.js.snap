// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown snapshot testing renders a list of options 1`] = `
<select
  className="t-select"
  onChange={[Function]}
  value="10"
>
  <option
    className="t-option"
    value="10"
  >
    Del Piero
  </option>
  <option
    className="t-option"
    value="21"
  >
    Dybala
  </option>
</select>
`;
